### DISCUSSION TITLE
Vulnerable alerts handling stops working on GHE

### DISCUSSION THREAD

**Discussion by pofengliu** [renovatebot/renovate#15907] [Request Help] [answered]
### How are you running Renovate?
Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.
31.53.2, 32.66.5, 32.74.3

### Please select which platform you are using if self-hosting.
GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.
GitHub Enterprise Server 3.3.7

### Describe the bug
Renovate-bot reports vulnerable alerts found with version 31.53.2
```
"msg":"Performing Github GraphQL request", ...
"url":"https://git.linecorp.com/api/graphql","msg":"Applying Bearer authentication", ...
"url":"https://git.linecorp.com/api/graphql","options":{"method":"post", ...
"alerts":[{"dismissReason":null,"vulnerableManifestFilename":"package.json", ...
"alerts":{"npm/ejs":{"< 3.1.7":"3.1.7"}},"msg":"GitHub vulnerability details", ...
"alertPackageRules":[{"matchDatasources":["npm"],"matchPackageNames":["ejs"], ...
"msg":"findIssue(Dependency Dashboard)", ...
```
Renovate-bot reports vulnerable alerts NOT found with version 32.66.5, 32.74.3
```
"msg":"Performing Github GraphQL request", ...
"url":"https://git.linecorp.com/api/graphql","msg":"Applying Bearer authentication", ...
"url":"https://git.linecorp.com/api/graphql","options":{"method":"post", ...
"msg":"No vulnerability alerts found", ...
"msg":"No vulnerability alerts found", ...
"msg":"findIssue(Dependency Dashboard)", ...
```
The same git repo, the same docker CLI command and config to run renovate-bot, the only difference is the docker image version, but alerts handling behavior are different.
I read similar issues [#15561](https://github.com/renovatebot/renovate/discussions/15561) and [#14310](https://github.com/renovatebot/renovate/discussions/14310) that Vulnerability Alerts are supported in GitHub only. 
Pls advise that does it include GHE or not ?? 
Due to the code structure has changed and I am not be able to easily compare what was updated since then.

**Comment by rarkins** [comment]
In theory it should work with both, so anything short of that is a bug we introduced accidentally (more likely) or due to a change in GHE API we didn't notice (less likely). Always a chance that it's related to your environment, but you seem to have some adequate troubleshooting to decrease that likelihood.

**Comment by pofengliu** [comment]
I pull down the latest main of renovatebot/renovate and run debug. I got below error but somehow it does not get thrown out. Does the message make sense to you?? 
"Field 'vulnerabilityAlerts' doesn't accept argument 'states'"
![github_ts_—_renovate](https://user-images.githubusercontent.com/2379575/172146826-8a03c2f2-d5e8-4c1e-9c06-61ea2ccc8f49.png)

**Reply by rarkins** [reply]
Your GHE doesn't support the states filter for graphQL. What GHE version are you using

**Reply by pofengliu** [reply]
GitHub Enterprise Server 3.3.7

**Comment by rarkins** [comment, marked as answer]
Let's continue the discussion in #15916
